src/view_3d.cpp

```diff
@@ -242,23 +250,49 @@ void view_3d::pick_object(level& lvl, ImVec2 position) {
 	glFinish();
 	glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
 	
-	unsigned char coded_object[4];
-	glReadPixels(position.x, position.y, 1 , 1, GL_RGBA, GL_UNSIGNED_BYTE, coded_object);
+	constexpr int selectX = 10, selectY = 10;
+	constexpr int size = selectX * selectY;
+	
+	unsigned char buffer[size*4];
+	glReadPixels(position.x, position.y, selectX , selectY, GL_RGBA, GL_UNSIGNED_BYTE, buffer);
 	
-	object_type type = static_cast<object_type>(coded_object[0]);
-	uint16_t index = coded_object[1] + (coded_object[2] << 8);
+	// TODO: make this go from the center and not from the left
+	// explanation: glReadPixels reads a square from the lower left corner. We want to
+	// check from the middle of the rectangle, to make it better for the user. This
+	// works for now though, and lets you select the extremely small splines with less precision.
+
+	struct {
+		object_type type;
+		unsigned char coded_object[3];
+	} selected_object;
+
+	bool found = false;
+	for(int i = 0; i < size; i+=4) {
+		if (buffer[i+selectY*i] > 0) {
+			selected_object = { static_cast<object_type>(buffer[i+selectY*i]), buffer[1+i+selectY*i] };
+			found = true;
+			break;
+		}
+	}
+
```

Comment:
I don't know exactly what this code is doing, but it seems to be selecting the wrong object. I think a better way to improve the selection would just be to make e.g. the splines render thicker in when drawing the pickframe with e.g. glLineWidth.

Reply:
Also all the splines seem to have a mysterious new point at the origin now.

![Screenshot_20200404_220321](https://user-images.githubusercontent.com/43898262/78461368-27b20d00-76c0-11ea-808c-116c812663ea.png)